sdk/src/Core/Amazon.Runtime/Internal/Auth/BearerTokenSigner.cs

```diff
@@ -13,66 +13,47 @@
  * permissions and limitations under the License.
  */
 
-using System;
 #if AWS_ASYNC_API
 using System.Threading;
 using System.Threading.Tasks;
 #endif
-using Amazon.Runtime.Internal;
-using Amazon.Runtime.Internal.Auth;
 using Amazon.Runtime.Internal.Util;
 
-namespace Amazon.Runtime
+namespace Amazon.Runtime.Internal.Auth
 {
     public class BearerTokenSigner : AbstractAWSSigner
     {
         public override bool RequiresCredentials { get; } = false;
 
-#if BCL
+        /// <summary>
+        /// The resolved bearer token, determined during the auth resolver handler.
+        /// </summary>
+        public string ResolvedToken { get; internal set; }
+
         public override void Sign(
             IRequest request, 
             IClientConfig clientConfig, 
             RequestMetrics metrics, 
             string awsAccessKeyId,
             string awsSecretAccessKey)
         {
-            if (request.Endpoint.Scheme == "http")
-            {
-                throw new AmazonClientException(
-                    $"The configured endpoint [{request.Endpoint}] is invalid for the bearer authorization scheme. " +
-                    "Endpoint must not use 'http'.");
-            }
-
-            if (null == clientConfig.AWSTokenProvider)
-                return;
-
-            if (!clientConfig.AWSTokenProvider.TryResolveToken(out var token) ||
-                string.IsNullOrEmpty(token.Token))
-            {
-                throw new AmazonClientException("No Token found.  Operation requires a Bearer token.");
-            }
-
-            request.Headers["Authorization"] = $"Bearer {token}";
-        }
-#else
-        public override void Sign(
-            IRequest request,
-            IClientConfig clientConfig,
-            RequestMetrics metrics,
-            string awsAccessKeyId,
-            string awsSecretAccessKey)
-        {
-            throw new NotSupportedException($"Use {nameof(SignAsync)} instead.");
+            InternalSign(request);
         }
-#endif
 
 #if AWS_ASYNC_API
-        public override async Task SignAsync(
+        public override Task SignAsync(
             IRequest request, 
             IClientConfig clientConfig, 
             RequestMetrics metrics, 
             ImmutableCredentials credentials,
```

Comment:
How much would that break things if we change the `credentials` parameter to `BaseIdentity` and then each signer cast the credentials to the type it expects. I really don't like having this `ResolvedToken` instance state for an object that is going to be used multi threaded.

Reply:
I updated the signer method signature in https://github.com/aws/aws-sdk-net/pull/3562/commits/32ab9122a184a2403d6176c675b3389b00b107ce (it did require changing more code than I expected, but I removed this instance state).